ws-nextjs-app/pages/[service]/send/[id]/FormContext/index.tsx

```diff
@@ -107,14 +107,20 @@ export const FormContextProvider = ({
 
   const handleSubmit = async (event: FormEvent) => {
     event.preventDefault();
-    setSubmitted(true);
     setAttemptedSubmit(true);
 
-    // Reset error state
-    setSubmissionError(null);
+    const validatedFormData = validateFormState(formState);
+    setFormState(validatedFormData);
+
+    const formInvalidErrors = Object.values(validatedFormData).filter(
+      item => item.isValid === false,
+    ).length;
 
-    // Validate
-    setFormState(state => validateFormState(state));
+    if (formInvalidErrors > 0) return;
+
+    setSubmitted(true); // check placement
```

Comment:
This is repsonsible for changing the 'Submit' button to 'Loader'. (Which is probably some logic we want to change/ remove).